chore(aws-lambda-connector) add docs (#1233)
camunda/camunda-docs: docs/components/integration-framework/connectors/out-of-the-box-connectors/aws-lambda.md

---
id: awslambda
title: AWS Lambda Connector

> description: Invoke aws lambda function.

Reviewer: ```suggestion
description: Invoke AWS Lambda function.
```

> description: Invoke AWS Lambda function.

Reviewer 2: ```suggestion
description: Invoke AWS Lambda functions.
```

---

The AWS Lambda Connector allows you to connect your BPMN service with the Amazon Web Service's [AWS Lambda Service](https://aws.amazon.com/lambda/).

## Prerequisites

> To use an **AWS Lambda Connector**, you need to have an [AWS Lambda Function](https://docs.aws.amazon.com/lambda/latest/dg/welcome.html), IAM key, and secret pair with permissions for execute function. See the [AWS Lambda developer guide](https://docs.aws.amazon.com/lambda/latest/dg/lambda-permissions.html) to learn more.

Reviewer 3: "...and secret pair with permissions for execute function" -- do you mean "...and secret pair with permissions for the execute function" or "to execute the function?"

> To use an **AWS Lambda Connector**, you need to have an [AWS Lambda Function](https://docs.aws.amazon.com/lambda/latest/dg/welcome.html), IAM key, and secret pair with permissions for execute function. See the [AWS Lambda developer guide](https://docs.aws.amazon.com/lambda/latest/dg/lambda-permissions.html) to learn more.

Reviewer 2: ```suggestion
To use the **AWS Lambda Connector**, you need to have an [AWS Lambda Function](https://docs.aws.amazon.com/lambda/latest/dg/welcome.html), IAM key and secret pair with permissions to execute the function. See the [AWS Lambda developer guide](https://docs.aws.amazon.com/lambda/latest/dg/lambda-permissions.html) to learn more.
```

Reviewer 2: @igpetrov @Oleksiivanov, do we need to add this sentence in the Prerequisites?

It is highly recommended not to expose your AWS IAM credentials as plain text but rather use Camunda Secrets. See an [appendix entry](#how-do-i-store-aws-iam-secrets-for-my-lambda-connector) to find out more.

Reviewer 2: @Oleksiivanov this looks good, but it can just be normal text. So remove italics and bold. 
Also last sentence should be "See Manage secrets to find out more."

> To use an **AWS Lambda Connector**, you need to have an [Aws Lambda Function](https://docs.aws.amazon.com/lambda/latest/dg/welcome.html) and IAM key and secret pair with permissions for execute function. See the [AWS Lambda developer guide](https://docs.aws.amazon.com/lambda/latest/dg/lambda-permissions.html) to learn more.

Reviewer: ```suggestion
To use an **AWS Lambda Connector**, you need to have an [AWS Lambda Function](https://docs.aws.amazon.com/lambda/latest/dg/welcome.html) and IAM key and secret pair with permissions for execute function. See the [AWS Lambda developer guide](https://docs.aws.amazon.com/lambda/latest/dg/lambda-permissions.html) to learn more.
```

## Create an AWS Lambda Connector task

> To use an **AWS Lambda Connector** in your process, either change the type of existing task by clicking on it and using the wrench-shaped **Change type** context menu icon, or create a new Connector task by using the **Append Connector** context menu. Follow our [guide on using Connectors](../use-connectors.md) to learn more.

Reviewer 2: ```suggestion
To use the **AWS Lambda Connector** in your process, either change the type of existing task by clicking on it and using the wrench-shaped **Change type** context menu icon, or create a new Connector task by using the **Append Connector** context menu. Follow our [guide on using Connectors](../use-connectors.md) to learn more.
```

> ## Make your AWS Lambda Connector for invoke function

Reviewer 2: ```suggestion
## Make your AWS Lambda Connector for invoking functions executable
```

To make the **AWS Lambda Connector** executable, fill out the mandatory fields highlighted in red in the properties panel:

![AWS Lambda Filled](../img/connectors-aws-lambda-filled.png)

1. Set the relevant IAM key and secret pair in the **Authentication** section. For example, `secrets.MY_AWS_ACCESS_KEY`. The value can be plain text, but this is not recommended due to security concerns.

> 2. Set the relevant AWS region in the **Authentication** section. See the [Regions and Zones](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html) to learn more.

Reviewer 2: ```suggestion
2. Set the relevant AWS region in the **Authentication** section. See [Regions and Zones](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html) to learn more.
```

> 3. In the **Select Operation** section, operation type is set to synchronized invoke function; asynchronous call type is still in development.

Reviewer 2: ```suggestion
3. In the **Select Operation** section, the default option is set to synchronous invocation; an asynchronous invocation option is currently not available.
```

> 4. In the **Operation Details** section, fill the field **Function name**. This field can be [function url](https://docs.aws.amazon.com/lambda/latest/dg/lambda-urls.html?icmpid=docs_lambda_help), [function ARN](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html), function name, or alias.

Reviewer 2: ```suggestion
4. In the **Operation Details** section, fill the field **Function name**. This field can be a [function url](https://docs.aws.amazon.com/lambda/latest/dg/lambda-urls.html?icmpid=docs_lambda_help), [function ARN](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html), function name, or an alias.
```

> 5. (Optional) Feel **Payload** field in the **Operation Details** section. Payload must be in json format - this is the data that will be processed by your lambda function. This field requires FEEL input.

Reviewer: ```suggestion
5. (Optional) Feel **Payload** field in the **Operation Details** section. Payload must be in JSON format - this is the data that will be processed by your Lambda function. This field requires FEEL input.
```

> 5. (Optional) FEEL **Payload** field in the **Operation Details** section. Payload must be in JSON format as this is the data that will be processed by your Lambda function. This field requires FEEL input.

Reviewer 2: ```suggestion
5. (Optional) The **Payload** field in the **Operation Details** section is optional. This field requires FEEL input. Payload must be in JSON format as this is the data that will be processed by your Lambda function. 
```

Reviewer 2: @igpetrov @Oleksiivanov, should we provide an example?

## AWS Lambda Connector response

The **AWS Lambda Connector** returns the HTTP status code, executed version, and payload (the response from the function, or an error object).
The following fields are available in the response variable:

- `statusCode` - HTTP status code.

> - `executedVersion` - Executed version of Lambda function.

Reviewer 2: ```suggestion
- `executedVersion` - Executed version of the Lambda function.
```

---
id: aws-lambda
title: AWS Lambda Connector
description: Invoke AWS Lambda functions.
---

The AWS Lambda Connector allows you to connect your BPMN service with the Amazon Web Service's [AWS Lambda Service](https://aws.amazon.com/lambda/).

## Prerequisites

> To use an **AWS Lambda Connector**, you need to have an [AWS Lambda Function](https://docs.aws.amazon.com/lambda/latest/dg/welcome.html), IAM key, and secret pair with permissions for executing function. See the [AWS Lambda developer guide](https://docs.aws.amazon.com/lambda/latest/dg/lambda-permissions.html) to learn more.

Reviewer 2: ```suggestion
To use the **AWS Lambda Connector**, you need to have an [AWS Lambda Function](https://docs.aws.amazon.com/lambda/latest/dg/welcome.html), IAM key and secret pair with permissions for executing function. See the [AWS Lambda developer guide](https://docs.aws.amazon.com/lambda/latest/dg/lambda-permissions.html) to learn more.
```

It is highly recommended not to expose your AWS IAM credentials as plain text but rather use Camunda Secrets. 

> See [Manage secrets](../img/connectors-aws-lambda-filled.png) find out more.

Reviewer 2: ```suggestion
See [Manage secrets](../img/connectors-aws-lambda-filled.png) to find out more.
```
